superx/models/__init__.py

```diff
@@ -26,7 +26,7 @@ class Branch(db.Model):
     name = Column(Text)
     address = Column(Text)
     sub_chain_id = Column(Integer)
-    chain_id = Column(db.ForeignKey('chain.id'))
+    chain_id = Column(db.ForeignKey('chain.id'), primary_key=True)
```

Comment:
I think that chin_id can't be primary because the primary key must be unique for each entry in the table.

Reply:
This is a composite primary key with 2 fields.
I recommend making the primary key an auto increment number and changing  chain_id,branch_id to be a unique constraint.